mmseg/models/backbones/mit.py

```diff
@@ -0,0 +1,434 @@
+
+
+class PEConv(BaseModule):
+    """Mix-FFN use 3x3 depth-wise Conv to provide positional encode
+    information.
+
+    Args:
+        embed_dims (int): The channels of token after embedding.
+        kernel_size (int): The kernel size of convolution operation.
+            Default: 3.
+        stride (int): The kernel slide move distance of one step.
+            Default: 1.
+    """
+
+    def __init__(self, embed_dims, kernel_size=3, stride=1):
+        super().__init__()
+        self.conv = ConvModule(
+            in_channels=embed_dims,
+            out_channels=embed_dims,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=(kernel_size - 1) // 2,
+            bias=True,
+            groups=embed_dims,
+            norm_cfg=None,
+            act_cfg=None)
+
+    def forward(self, x):
+
+        x = self.conv(x)
+
+        return x
+
+
+class MixFFN(BaseModule):
+    """An implementation of MixFFN of Segformer.
+
+    The differences between MixFFN & FFN:
+        1. Use 1X1 Conv to replace Linear layer.
+        2. Introduce 3X3 Conv to encode positional information.
+
+    Args:
+        embed_dims (int): The feature dimension. Same as
+            `MultiheadAttention`. Defaults: 256.
+        feedforward_channels (int): The hidden dimension of FFNs.
+            Defaults: 1024.
+        num_fcs (int, optional): The number of fully-connected layers in
+            FFNs. Default: 2.
+        act_cfg (dict, optional): The activation config for FFNs.
+            Default: dict(type='ReLU')
+        ffn_drop (float, optional): Probability of an element to be
+            zeroed in FFN. Default 0.0.
+        dropout_layer (obj:`ConfigDict`): The dropout_layer used
+            when adding the shortcut.
+        add_identity (bool, optional): Whether to add the
+            identity connection. Default: `True`.
+        init_cfg (obj:`mmcv.ConfigDict`): The Config for initialization.
+            Default: None.
+    """
+
+    def __init__(self,
+                 embed_dims,
+                 feedforward_channels,
+                 num_fcs=2,
+                 act_cfg=dict(type='GELU'),
+                 ffn_drop=0.,
+                 pe_index=0.,
+                 dropout_layer=None,
+                 add_identity=True,
+                 init_cfg=None):
+        super(MixFFN, self).__init__(init_cfg)
+        assert num_fcs >= 2, 'num_fcs should be no less ' \
+            f'than 2. got {num_fcs}.'
+
+        self.embed_dims = embed_dims
+        self.feedforward_channels = feedforward_channels
+        self.num_fcs = num_fcs
+        self.act_cfg = act_cfg
+        self.activate = build_activation_layer(act_cfg)
+
+        conv1x1 = partial(
+            ConvModule,
+            kernel_size=1,
+            stride=1,
+            bias=True,
+            norm_cfg=None,
+            act_cfg=None)
+
+        layers = []
+        in_channels = embed_dims
```

Comment:
`pe_index` belongs to [0, num_fcs].
`pe_index=0` --> before the first `1x1 conv` or `FC`.
`pe_index=num_fcs` --> after the last `1x1 conv` or `FC`.